### PR TITLE
Add PlatformCheckbox

### PR DESCRIPTION
**Pull request by bookshiyi**

### CODE REVIEW

File: lib/src/platform_checkbox.dart [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    this.tristate = false,
+    required this.onChanged,
+    this.activeColor,
+    this.checkColor,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+    this.shape,
+    this.side,
+  }) : assert(tristate || value != null);
```
Comment (aqwert) [inline lib/src/platform_checkbox.dart:28]: Need to remove this assert

File: lib/src/platform_checkbox.dart [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    this.checkColor,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+    this.shape,
+    this.side,
+    //Platform
+    this.material,
+    this.cupertino,
+  }) : assert(tristate || value != null);
```
Comment (aqwert) [inline lib/src/platform_checkbox.dart:134]: Need to remove this assert. See below why

File: lib/src/platform_checkbox.dart [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+  @override
+  Checkbox createMaterialWidget(BuildContext context) {
+    final data = material?.call(context, platform(context));
```
Comment (aqwert) [inline lib/src/platform_checkbox.dart:138]: Need to put the assert here that combines the properties as a result from the widget props and the data props.

```
final tristate = data?.tristate ?? this.tristate;
final value = data?.value ?? this.value;
assert(tristate || value != null);

```

File: lib/src/platform_checkbox.dart [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    return Checkbox(
+      //Common
+      key: data?.widgetKey ?? widgetKey,
+      value: data?.value ?? value,
+      tristate: data?.tristate ?? tristate,
```
Comment (aqwert) [inline lib/src/platform_checkbox.dart:143]:
```suggestion
      value: value,
      tristate: ristate,
```

File: lib/src/platform_checkbox.dart [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      splashRadius: data?.splashRadius,
+      materialTapTargetSize: data?.materialTapTargetSize,
+      visualDensity: data?.visualDensity,
+      isError: data?.isError ?? false,
+    );
+  }
+
+  @override
+  CupertinoCheckbox createCupertinoWidget(BuildContext context) {
+    final data = cupertino?.call(context, platform(context));
```
Comment (aqwert) [inline lib/src/platform_checkbox.dart:166]: Need to do the same here in the same way for material for adding the assert

**Review by aqwert — CHANGES_REQUESTED** [review]

**Comment by aqwert** [comment]
Since the previous PR was merged, could you rebase to fix the conflicts. Thanks

**Comment by bookshiyi** [comment]
I have resolved the conflict.

**Pull request merged**